third_party/nvfuser/csrc/scheduler/normalization.cpp

```diff
@@ -1221,4 +1515,257 @@ void schedulePersistentKernel(Fusion* fusion, const ReductionParams& rparams) {
+    partialResult->cacheBefore();
+    partialResult->setMemoryType(MemoryType::Global);
+    TensorView* partialResultReload = partialResult->cacheAfter();
+
+    boundaryNodesSet.insert(partialResultReload);
+    cached_gmem.emplace_back(partialResult);
+    cached_gmem_reload.emplace_back(partialResultReload);
+
+    if (rparams.multiple_reds_per_blk) {
+      if (rparams.tidx_for_outer_reduction) {
+        outer_reduction_tv->split(
+            0, NamedScalar::getParallelDim(ParallelType::TIDx));
+        outer_reduction_tv->axis(1)->parallelize(ParallelType::TIDx);
+        // to use warp reduction
+        if (rparams.pad_outer_reduction_to_warp) {
+          outer_reduction_tv->axis(1)->padToMultipleOfWarp();
+        }
+      } else {
+        outer_reduction_tv->split(
+            0, NamedScalar::getParallelDim(ParallelType::TIDy));
+        outer_reduction_tv->axis(1)->parallelize(ParallelType::TIDy);
+      }
+
+      // iteration domain
+      int axisID = -1;
+      if (rparams.vectorization_factor_tmp_gmem_read > 1) {
+        outer_reduction_tv->split(
+            axisID, rparams.vectorization_factor_tmp_gmem_read);
+        outer_reduction_tv->axis(axisID--)->parallelize(ParallelType::Unroll);
+      }
+
+      if (rparams.tidx_for_outer_reduction) {
+        outer_reduction_tv->split(
+            axisID, NamedScalar::getParallelDim(ParallelType::TIDy));
+        outer_reduction_tv->axis(axisID--)->parallelize(ParallelType::TIDy);
+      } else {
+        outer_reduction_tv->split(
+            axisID, NamedScalar::getParallelDim(ParallelType::TIDx));
+        outer_reduction_tv->axis(axisID--)->parallelize(ParallelType::TIDx);
+      }
+
+      outer_reduction_tv->split(
+          axisID, NamedScalar::getParallelDim(ParallelType::BIDy));
+      outer_reduction_tv->axis(axisID--)->parallelize(ParallelType::BIDy);
+    } else {
+      // reduction domain
+      outer_reduction_tv->split(
+          0, NamedScalar::getParallelDim(ParallelType::TIDy));
+      outer_reduction_tv->axis(1)->parallelize(ParallelType::TIDy);
+      // iteration domain
+      int axisID = -1;
+      if (rparams.vectorization_factor_tmp_gmem_read > 1) {
+        outer_reduction_tv->split(
+            axisID, rparams.vectorization_factor_tmp_gmem_read);
+        outer_reduction_tv->axis(axisID--)->parallelize(ParallelType::Unroll);
+      }
+      if (rparams.lparams.bdimx() > 1) {
+        outer_reduction_tv->split(
+            axisID, NamedScalar::getParallelDim(ParallelType::TIDx));
+        outer_reduction_tv->axis(axisID--)->parallelize(ParallelType::TIDx);
+      }
+
+      outer_reduction_tv->split(
+          axisID, NamedScalar::getParallelDim(ParallelType::BIDy));
+      outer_reduction_tv->axis(axisID--)->parallelize(ParallelType::BIDy);
+    }
+
+    auto outer_reference_tv =
+        reduction_scheduler_utils::sortAndRFactor(outer_reduction_tv);
+    outer_reference_tvs.emplace_back(outer_reference_tv);
+  }
```

Comment:
Please factor this out as a function to schedule the outer part of the reductions

Reply:
Also the inner scheduler part. So this function should look like:

```
void schedulePersistentKernelInnerOuter(...) {
  scheduleCombinedInner();
  scheduleCombinedOuter();
  do_final_common_scheduling;
}
```

Reply:
refactored both `schedulePersistentKernelInnerOuter` and `schedulePersistentKernel`